src/components/flowchart/flowchart.js

```diff
@@ -451,32 +495,91 @@ export class FlowChart extends Component {
+      onClickToExpandModularPipeline(id);
+    } else {
+      this.handleSingleNodeClick(node);
+
+      // the hold shift only happens on clicking a node first
+      if (event.shiftKey) {
+        this.handleShiftClick(node);
+      }
+    }
+
+    event.stopPropagation();
+  };
+
+  handleSingleNodeClick = (node) => {
+    const { id } = node;
     const {
-      onClickToExpandModularPipeline,
       displayMetadataPanel,
       onLoadNodeData,
-      toSelectedNode,
       onToggleNodeClicked,
+      toSelectedNode,
     } = this.props;
 
-    if (type === 'modularPipeline') {
-      onClickToExpandModularPipeline(id);
+    displayMetadataPanel ? onLoadNodeData(id) : onToggleNodeClicked(id);
+    toSelectedNode(node);
+
+    const { from, to, range } = this.state.slicedPipelineState;
+    // if both "from" and "to" are defined
+    // then on a single node click, it should reset the sliced pipeline state
+    if (from !== null && to !== null) {
+      this.updateSlicedPipelineState(null, null, []);
     } else {
-      displayMetadataPanel ? onLoadNodeData(id) : onToggleNodeClicked(id);
-      toSelectedNode(node);
+      // Else, set the first node as the 'from' node based on current state
+      // we need this so that if user hold shift and click on a second node,
+      // the 'from' node is already set
+      this.updateSlicedPipelineState(id, to, range);
     }
-    event.stopPropagation();
+
+    // Reset the slicePipeline on single node click
+    this.props.onSlicePipeline(null, null);
+  };
+
+  handleShiftClick = (node) => {
+    // Close meta data panel
+    this.props.onLoadNodeData(null);
+
+    const fromNodeId = this.state.slicedPipelineState.from || node.id;
+    const toNodeId = node.id;
+
+    const newState = {
+      ...this.state.slicedPipelineState,
+      from: fromNodeId,
+      to: toNodeId,
+    };
+
+    this.setState({ slicedPipelineState: newState });
+
+    this.props.onSlicePipeline(fromNodeId, toNodeId);
+    this.props.onApplySlice(false);
   };
 
   /**
    * Remove a node's focus state and dim linked nodes
    */
-  handleChartClick = () => {
+  handleChartClick = (event) => {
     if (this.props.clickedNode) {
       this.props.onLoadNodeData(null);
       // To reset URL to current active pipeline when click outside of a node on flowchart
       this.props.toSelectedPipeline();
     }
+
+    // Determine if the click event occurred on the slice button.
+    const isSliceButtonClicked =
+      this.slicedPipelineActionRef.current &&
+      this.slicedPipelineActionRef.current.contains(event.target);
+
+    // Check if the pipeline is sliced, no slice button is clicked, and no filters are applied
+    if (this.props.slicedPipeline && !isSliceButtonClicked) {
+      this.props.onResetSlicePipeline();
+      this.updateSlicedPipelineState(null, null, []);
+      // To reset URL to current active pipeline when click outside of a node on flowchart
+      this.props.toSelectedPipeline();
```

Comment:
On click on Slice button nothing is happening. Is it expected just for this PR ?

Reply:
yeah, no functionality for the slice button in this PR. it will be in a separate PR